File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Love.java
package com.xebia.functional.xef.java.auto;

import java.util.List;
import java.util.concurrent.ExecutionException;

public class Love {
    public List<String> loveList;
    public static void main(String[] args) throws ExecutionException, InterruptedException {
        try (AIScope scope = new AIScope()) {
            scope.prompt("tell me you like me with just emojis", Love.class)
                  .thenAccept(love -> System.out.println(love.loveList))
                  .get();
        }
    }
}

File: example/src/main/kotlin/com/xebia/functional/langchain4k/auto/Love.kt
package com.xebia.functional.langchain4k.auto

import com.xebia.functional.auto.prompt
import com.xebia.functional.auto.getOrElse

suspend fun main() {
  prompt {
    val love: List<String> = prompt("tell me you like me with just emojis")
    println(love)
  }.getOrElse { println(it) }
}
